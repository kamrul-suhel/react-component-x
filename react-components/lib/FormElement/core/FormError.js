"use strict";

Object.defineProperty(exports, "__esModule", {
	value: true
});
exports.default = FormError;

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function FormError(props) {

	if (props.valid || !props.errors || props.errors.length === 0) {
		return null;
	}

	return _react2.default.createElement(
		"div",
		{ className: "form-field-error-wrapper" },
		_react2.default.createElement(
			"ul",
			null,
			props.errors.map(function (error, i) {
				return _react2.default.createElement(
					"li",
					{ key: props.name + "error" + i },
					error
				);
			})
		)
	);
}