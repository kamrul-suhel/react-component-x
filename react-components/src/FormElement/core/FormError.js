import React from 'react';

export default function FormError(props) {

	if (props.valid || !props.errors || props.errors.length === 0) {
		return null;
	}

	return (
		<div className="form-field-error-wrapper">
			<ul>
				{props.errors.map((error, i) => <li key={`${props.name}error${i}`}>{error}</li>)}
			</ul>
		</div>
	);
}
