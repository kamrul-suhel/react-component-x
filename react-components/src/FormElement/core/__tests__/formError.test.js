import React from 'react';
import renderer from 'react-test-renderer';
import FormError from '../FormError';

it('renders correctly', () => {
	const tree = renderer.create(<FormError valid={false} errors={['Something went wrong']} />).toJSON();
	expect(tree).toMatchSnapshot();
});
