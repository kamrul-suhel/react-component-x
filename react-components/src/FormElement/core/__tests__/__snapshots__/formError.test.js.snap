// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="form-field-error-wrapper"
>
  <ul>
    <li>
      Something went wrong
    </li>
  </ul>
</div>
`;
